exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DefaultSqlHandler.java

```diff
@@ -459,7 +468,13 @@ protected Prel convertToPrel(RelNode drel) throws RelConversionException, SqlUns
     /* The order of the following transformations is important */
 
     /*
-     * 0.) For select * from join query, we need insert project on top of scan and a top project just
+     * 0.)
+     * Add top project before screen operator or writer to ensure that final output column names are preserved.
+     */
+    phyRelNode = TopProjectVisitor.insertTopProject(phyRelNode, validatedRowType);
```

Comment:
Make sense to me that you put this Visitor handling first, after physical transform() call, before these existing visitor handling.